packages/components/src/internal/components/search/SampleFinderSection.tsx

```diff
@@ -205,17 +225,87 @@ export const SampleFinderSection: FC<Props> = memo(props => {
 
             api.query.incrementClientSideMetricCount(SAMPLE_FILTER_METRIC_AREA, 'filterModalApply');
         },
-        [filters, filterChangeCounter, onFilterEdit, onFilterDelete, chosenEntityType]
+        [filters, filterChangeCounter, onFilterEdit, onFilterDelete, chosenEntityType, cardDirty, currentView]
+    );
+
+    const loadSearch = useCallback(
+        async (view: FinderReport) => {
+            let cardJson = null;
+
+            if (view.isSession) cardJson = sessionStorage.getItem(getLocalStorageKey());
+            else if (view.reportId) cardJson = await loadFinderSearch(view);
+            if (!cardJson) return;
+
+            const finderSessionData = searchFiltersFromJson(cardJson);
+            const newFilters = finderSessionData.filters;
+            if (!newFilters) return;
+
+            updateFilters(filterChangeCounter + 1, newFilters, false, view.isSession);
+            setShowSaveViewDialog(false);
+            setCurrentView(view);
+        },
+        [filterChangeCounter]
+    );
+
+    const onSaveComplete = useCallback((view: FinderReport) => {
+            setShowSaveViewDialog(false);
+            setViewDirty(false);
+            setCurrentView(view);
+            setSavedViewChangeCounter(savedViewChangeCounter + 1)
+        },
+        [savedViewChangeCounter]
+    );
+
+    const saveSearch = useCallback(
+        async (saveCurrentName?: boolean) => {
+            if (saveCurrentName) {
+                const viewJson = JSON.stringify({ filters: getSearchFilterObjs(filters) });
+                await saveFinderSearch(currentView, viewJson, saveCurrentName);
+                setViewDirty(false);
+            } else {
+                setShowSaveViewDialog(true);
+            }
+        },
+        [currentView, filters]
+    );
+
+    const searchViewJson = useMemo(() => {
+        return JSON.stringify({
+            filters: getSearchFilterObjs(filters),
+        });
+    }, [showSaveViewDialog, filters]);
+
+    const manageSearches = useCallback(() => {
+        setShowManageViewsDialog(true);
+    }, []);
+
+    const onManageSearchesDone = useCallback(
+        (hasChange: boolean) => {
+            setShowManageViewsDialog(false);
+            if (hasChange) {
+                setSavedViewChangeCounter(savedViewChangeCounter + 1);
+            }
+        },
+        [savedViewChangeCounter]
     );
 
     return (
         <Section
-            title={SAMPLE_FINDER_TITLE}
-            caption={SAMPLE_FINDER_CAPTION}
+            title={
+                <SampleFinderSavedViewsMenu
+                    loadSearch={loadSearch}
+                    saveSearch={saveSearch}
+                    manageSearches={manageSearches}
+                    currentView={currentView}
+                    hasUnsavedChanges={viewDirty}
+                    sessionViewName={unsavedSessionViewName}
+                    key={filterChangeCounter + '-' + savedViewChangeCounter}
```

Comment:
since this key is changing with each `savedViewChangeCounter` change, it will re construct the `SampleFinderSavedViewsMenu` right? in that case, you shouldn't need to care about the `key` prop in `SampleFinderSavedViewsMenu` to be used by `useEffect`.